pkg/netpol/eval/internal/k8s/netpol.go

```diff
@@ -208,7 +208,18 @@ func (np *NetworkPolicy) ruleSelectsPeer(rulePeers []netv1.NetworkPolicyPeer, pe
 					return false, err
 				}
 				peerNamespace := peer.GetPeerNamespace()
-				peerMatchesNamespaceSelector = selector.Matches(labels.Set(peerNamespace.Labels))
+				// checking if a selector matches labels by peer type; since representative peers selectors may need special handling
+				if isRepresentativePod(peer) {
+					// representative peer's namespace labels may be inferred from a rule with special matchExpression requirements
+					// and also contains the representative ns name label which is not relevant for comparison
+					peerMatchesNamespaceSelector, err = SelectorMatchesRepresentativePeerLabels(selector, peerNamespace.Labels,
```

Comment:
can't we just keep a reference to the selector(s) from which this representative peer was created,  and consider a match only if this is the relevant selecotr, instead of implementing containment of selectors comparison?

Reply:
I think we still need this containment, some selectors may intersect / be equivalent on one way to other selectors 
i.e. a representative peer (built from one selector) may match two or more selectors actually and then the connection contains the ports of all the selectors. 
an example : `test_exposure_with_different_rules_6` (there are more examples like this too)